### PR TITLE
Merge remote-tracking branch 'upstream/stable' into merge_stable

### PR DESCRIPTION
**Pull request by wilzbach**
Conflicts:
	src/dmd/backend/cgobj.d

### CODE REVIEW

**Comment by rainers** [comment]
Thanks. This seems to be running into out-of-memory problems, though.

**Comment by rainers** [comment]
> This seems to be running into out-of-memory problems, though.

https://github.com/dlang/dmd/pull/10232/commits/f93ec135b7af7c249ad60e272467514a9d0891cf seems to be the offending commit. The introduced silent limit on recursion depth seems problematic to begin with.

**Comment by rainers** [comment]
> [f93ec13](https://github.com/dlang/dmd/commit/f93ec135b7af7c249ad60e272467514a9d0891cf) seems to be the offending commit. The introduced silent limit on recursion depth seems problematic to begin with.

@wilzbach I think I have a fix for that, but it's mostly reverting that commit. What would be the correct workflow: applying the change to stable first, then redo this merge to master PR? Or adding the fix to this PR?

**Comment by wilzbach** [comment]
I think stable first would be better as there might be a point release soon.

**Comment by rainers** [comment]
> I think stable first would be better as there might be a point release soon.

Fix is merged, can this be rebased/redone?

**Comment by wilzbach** [comment]
> Fix is merged

Awesome. Thanks!

> can this be rebased/redone?

Yes, just merged stable in again and rebased against newest master. There were conflicts in dtemplate.d, traits.d, and typesem.d.

File: src/dmd/backend/cgobj.d [hunk 1]
```diff
@@ -460,24 +460,12 @@ version (X86ASM)
...
-            mov     EAX,index - [ESP]       ;
-            mov     ECX,p - [ESP]           ;
-        }
-    else
-        asm nothrow
-        {
-            naked                           ;
-            mov     EAX,index - [ESP+4]     ;
-            mov     ECX,p - [ESP+4]         ;
-        }
```
Comment (wilzbach) [inline src/dmd/backend/cgobj.d]: Not sure about this change, but I think I merged this incorrectly. See e.g. https://github.com/dlang/dmd/pull/10045

Comment (rainers) [reply src/dmd/backend/cgobj.d]: Yeah, looks good.

**Comment by rainers** [comment]
Error on buildkite:
```
Could not probe the compiler version for "dmd". Toolchain requirements might be ineffective
  | Fetching libdparse 0.10.13 (getting selected version)...
  | Fetching stdx-allocator 2.77.5 (getting selected version)...
  | Invalid SemVer format: 0.00.0
  | posix.mak:616: recipe for target 'generated/linux/release/64/tests_extractor' failed
  | make[1]: *** [generated/linux/release/64/tests_extractor] Error 2
  | make[1]: Leaving directory '/var/lib/buildkite-agent/builds/ci-agent-b008268a-0cc9-4064-908e-0d5803176a33-8/dlang/dmd/phobos'
  | posix.mak:646: recipe for target 'std/array.betterc' failed
  | make: *** [std/array.betterc] Error 2
```
Could that be caused by https://github.com/dlang/dmd/pull/10242?

**Comment by wilzbach** [comment]
> Could that be caused by #10242?

Likely, though I wonder why it passed for that result. Let's see whether https://github.com/dlang/dmd/pull/10244 helps.

**Review by rainers — APPROVED** [review]

**Comment by thewilsonator** [comment]
Hmm, dlang bot seems to be asleep.

**Pull request merged**